src/main/java/appeng/api/util/CraftCompleteListener.java

```diff
@@ -0,0 +1,19 @@
+package appeng.api.util;
+
+import java.io.Serializable;
+
+@FunctionalInterface
+public interface CraftCompleteListener<A1, A2, A3> extends Serializable {
+
+    long serialVersionUID = 734594276097234589L;
+
+    /**
+     * Applies this function to the given arguments.
+     *
+     * @param a1 the first function argument
+     * @param a2 the second function argument
+     * @param a3 the third function argument
+     * @return the function result
+     */
+    void apply(A1 a1, A2 a2, A3 a3);
```

Comment:
Since this interface now is only for one particular purpose, removing type parameters and giving appropriate argument names would be better.